src/parser/jobs/nin/modules/TrickAttackUsage.tsx

```diff
@@ -0,0 +1,126 @@
+
+const MUDRAS: ActionKey[] = [
+	'TEN',
+	'TEN_KASSATSU',
+	'CHI',
+	'CHI_KASSATSU',
+	'JIN',
+	'JIN_KASSATSU',
+]
+
+export class TrickAttackUsage extends Analyser {
+	static override handle = 'taUsage'
+
+	@dependency private data!: Data
+	@dependency private downtime!: Downtime
+	@dependency private suggestions!: Suggestions
+
+	private mudraActions: number[] = []
+
+	private taCasts: number[] = []
+	private lostTime: number = 0
+	private gcdCount: number = 0
+	private castHook?: EventHook<Events['action']>
+
+	override initialise() {
+		this.mudraActions = MUDRAS.map(k => this.data.actions[k].id)
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.castHook = this.addEventHook(playerFilter.type('action'), this.onCast)
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.TRICK_ATTACK.id), this.onTrickAttack)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onCast(event: Events['action']) {
+		const action = this.data.getAction(event.action)
+		if (event.timestamp >= this.parser.fight.start_time && action && action.onGcd && this.mudraActions.indexOf(action.id) === -1) {
+			// Don't count the individual mudras as GCDs for this - they'll make the count screw if Suiton wasn't set up pre-pull
+			this.gcdCount++
+		}
+	}
+
+	private onTrickAttack(event: Events['action']) {
+		if (this.castHook != null) {
+			this.removeEventHook(this.castHook)
+			this.castHook = undefined
+		}
+
+		if (this.taCasts.length > 0) {
+			const lastCast = this.taCasts[this.taCasts.length - 1]
+			const taAvailable = lastCast + this.data.actions.TRICK_ATTACK.cooldown
+			const downtime = this.downtime.getDowntime(
+				this.parser.fflogsToEpoch(taAvailable),
+				this.parser.fflogsToEpoch(event.timestamp),
+			)
+			this.lostTime += Math.max((event.timestamp - taAvailable) - downtime, 0)
+		}
+
+		this.taCasts.push(event.timestamp)
+	}
+
+	private onComplete() {
+		if (this.taCasts.length > 0) {
+			const lastCast = this.taCasts[this.taCasts.length - 1]
+			// lostTime is only the time they were actually holding it off CD, but we want to add in the CD time of the final cast for
+			// calculating how many theoretical casts were lost. For example, 20s of holding + last cast 40s before the end of the fight
+			// would mean that they could've squeezed in an extra cast with perfect timing.
+			const lostCasts = Math.floor((this.lostTime + (this.parser.currentEpochTimestamp - lastCast)) / this.data.actions.TRICK_ATTACK.cooldown)
+			this.suggestions.add(new TieredSuggestion({
+				icon: this.data.actions.TRICK_ATTACK.icon,
+				content: <Trans id="nin.ta-usage.suggestions.missed.content">
+					Avoid holding <ActionLink {...this.data.actions.TRICK_ATTACK}/> for extended periods of time. It's typically ideal to use it as close to on cooldown as possible in order to keep it aligned with all the other raid buffs and personal burst windows, as well as maximizing the number of uses per fight.
```

Comment:
```suggestion
					Avoid holding <ActionLink action="TRICK_ATTACK"/> for extended periods of time. It's typically ideal to use it as close to on cooldown as possible in order to keep it aligned with all the other raid buffs and personal burst windows, as well as maximizing the number of uses per fight.
```

Reply:
TIL this syntax exists. I'll update to it, and use it going forward in my other Analyser migration branch that's currently in the works.